### PR TITLE
Add FreeIPA lookup plugins

### PR DESCRIPTION
**Pull request by rsuplina**

### CODE REVIEW

File: plugins/lookup/env_freeipa_domain.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        self.set_options(var_options=variables, direct=kwargs)    
+        try: 
+            results = []
+            for term in LookupBase._flatten(terms):
+                environment = Cdpy().environments.describe_environment(term)
+                freeipa_client_domain = environment['freeipa']['domain']
+                results = [{'domain': freeipa_client_domain}]
+
+            if self.get_option('detailed'):
+                for term in LookupBase._flatten(terms):
```
Comment (wmudge) [inline plugins/lookup/env_freeipa_domain.py:81]: Collapse into a single for loop

File: plugins/lookup/env_freeipa_fqdn.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    lookup: env_freeipa_fqdn
```
Comment (wmudge) [inline plugins/lookup/env_freeipa_fqdn.py:19]: Rename to `env_freeipa_hosts`

File: plugins/lookup/env_freeipa_fqdn.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+display = Display()
+class LookupModule(LookupBase):
+    def run(self, terms, variables=None, **kwargs):
+        self.set_options(var_options=variables, direct=kwargs)     
+        try:
+            results = []
+            for term in LookupBase._flatten(terms):
+                free_ipa_info =Cdpy().sdk.call(svc='environments', func='get_freeipa_status', environmentName=term)     
+
+            if self.get_option('detailed'):
```
Comment (wmudge) [inline plugins/lookup/env_freeipa_fqdn.py:81]: Move to be inline with the for loop

**Review by wmudge — CHANGES_REQUESTED** [review]

File: plugins/lookup/env_freeipa_domain.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    lookup: env_freeipa_domain
+    author: Ronald Suplina (@rsuplina) <rsuplina@cloudera.com>
+    short_description: Get infromation about Freeipa CDP domain and freeipa host private ips for selected Environment
+    description:
+        - Allows you to retrieve information about Freeipa Domain for a given CDP Public Cloud Environment.
+        - If the Environment is not found or is ambigious, the lookup will return an error.
```
Comment (wmudge) [inline plugins/lookup/env_freeipa_domain.py:24]: Add a line or two about where and how this information can be used, i.e. in `/etc/resolv.conf`

File: plugins/lookup/env_freeipa_fqdn.py (hunk 2 above)
Comment (rsuplina) [reply plugins/lookup/env_freeipa_fqdn.py:19]: Renamed

File: plugins/lookup/env_freeipa_domain.py (hunk 1 above)
Comment (rsuplina) [reply plugins/lookup/env_freeipa_domain.py:81]: Collapsed

File: plugins/lookup/env_freeipa_fqdn.py (hunk 3 above)
Comment (rsuplina) [reply plugins/lookup/env_freeipa_fqdn.py:81]: Moved inline

File: plugins/lookup/env_freeipa_domain.py (hunk 4 above)
Comment (rsuplina) [reply plugins/lookup/env_freeipa_domain.py:24]: Added Information about FreeIPA Role prerequisite

File: plugins/lookup/env_freeipa_domain.py [hunk 5]
```diff
@@ -54,37 +55,42 @@
...
-            if self.get_option('detailed'):
-                for term in LookupBase._flatten(terms):
-                    environment = Cdpy().environments.describe_environment(term)   
+                
+                if self.get_option('detailed'):
                     server_ips = environment['freeipa']['serverIP']
-                    results = [{'domain': freeipa_client_domain,'server_ips' : server_ips}]     
+                    results = [{'domain': freeipa_client_domain,'server_ips' : server_ips}]    
+                else:
+                    results = [{'domain': freeipa_client_domain}]
```
Comment (wmudge) [inline plugins/lookup/env_freeipa_domain.py:88]: Have `detailed=False` return a string, not a dict with a single key. Just test with `wantlist=True` as well as `wantlist=False` to confirm this operation.

File: plugins/lookup/env_freeipa_domain.py [hunk 6]
```diff
@@ -18,9 +18,10 @@
 DOCUMENTATION = '''
     lookup: env_freeipa_domain
     author: Ronald Suplina (@rsuplina) <rsuplina@cloudera.com>
-    short_description: Get infromation about Freeipa CDP domain and freeipa host private ips for selected Environment
+    short_description: Get information about FreeIPA CDP domain and FreeIPA host private ips for selected Environment
```
Comment (wmudge) [inline plugins/lookup/env_freeipa_domain.py:21]: > Get information about the FreeIPA domain and DNS server IP address(es) for the selected CDP Public Cloud Environment

File: plugins/lookup/env_freeipa_domain.py [hunk 7]
```diff
@@ -18,9 +18,10 @@
...
     description:
-        - Allows you to retrieve information about Freeipa Domain for a given CDP Public Cloud Environment.
+        - Allows you to retrieve information about FreeIPA Domain for a given CDP Public Cloud Environment.
+        - Provides information required for FreeIPA install Role prerequisite task of updating '/etc/resolv'
```
Comment (wmudge) [inline plugins/lookup/env_freeipa_domain.py:24]: > You can use these details to update client DNS, e.g. set up entries in `/etc/resolv.conf`.

File: plugins/lookup/env_freeipa_domain.py [hunk 8]
```diff
@@ -30,7 +31,7 @@
             required: True
         detailed:
             description:
-                -  Whether to return private ip's of Freeipa hosts for selected CDP Public Cloudera Environment.
+                -  Whether to return private ip's of FreeIPA hosts for selected CDP Public Cloudera Environment.
```
Comment (wmudge) [inline plugins/lookup/env_freeipa_domain.py:34]: > Flag to return the IP address of each FreeIP host for a selected CDP Public Cloud Environment.

File: plugins/lookup/env_freeipa_domain.py [hunk 9]
```diff
@@ -41,11 +42,11 @@
 
 
 EXAMPLES = '''
-- name: Retrieve the Freeipa domain for a single CDP Public Cloud Environment
+- name: Retrieve the FreeIPA domain for a single CDP Public Cloud Environment
   ansible.builtin.debug:
     msg: "{{ lookup('cloudera.cloud.env_freeipa_domain', 'example-env') }}"
 
-- name: Retrieve the Freeipa domain and Private Freeipa host ips for a CDP Public Cloud Environment
+- name: Retrieve the FreeIPA domain and Private FreeIPA host ips for a CDP Public Cloud Environment
```
Comment (wmudge) [inline plugins/lookup/env_freeipa_domain.py:49]: > Retrieve the FreeIPA domain and host IP addresses for a CDP Public Cloud Environment

File: plugins/lookup/env_freeipa_domain.py [hunk 10]
```diff
@@ -54,37 +55,42 @@
 
 RETURN = '''
   _list:
-    description: List of Freeipa domains for selected Environments
+    description: List of FreeIPA domains for selected Environments
     type: list
     elements: list
```
Comment (wmudge) [inline plugins/lookup/env_freeipa_domain.py:60]: > elements: complex

Because it returns a list if `detailed=True` or strings otherwise.

**Review by wmudge — CHANGES_REQUESTED** [review]

File: plugins/lookup/env_freeipa_domain.py (hunk 6 above)
Comment (rsuplina) [reply plugins/lookup/env_freeipa_domain.py:21]: Changed

File: plugins/lookup/env_freeipa_domain.py (hunk 7 above)
Comment (rsuplina) [reply plugins/lookup/env_freeipa_domain.py:24]: Changed

File: plugins/lookup/env_freeipa_domain.py (hunk 8 above)
Comment (rsuplina) [reply plugins/lookup/env_freeipa_domain.py:34]: Changed

File: plugins/lookup/env_freeipa_domain.py (hunk 10 above)
Comment (rsuplina) [reply plugins/lookup/env_freeipa_domain.py:60]: Changed

**Review by wmudge — APPROVED** [review]

**Pull request merged**